lib/seattleflu/id3c/cli/command/etl/redcap_det_scan.py

```diff
@@ -888,8 +881,17 @@ def cast_to_string(string: str) -> Optional[str]:
         return None
 
     def cast_to_integer(string: str) -> Optional[int]:
+        value: Union[str, float] = string
```

Comment:
Would it be better to set the parameter `string`'s type hint to `Union[str, float]` rather than setting it here? (maybe also renaming `string`)

Reply:
We expect the argument to be a str, so i think we should keep it as such. 

I had to create a separate variable with the different type hint to avoid a typing error when I reassign the value with `age_ceiling()`:
```
error: Incompatible types in assignment (expression has type "float", variable has type "str")
```

Reply:
Ah, sorry. I didn't dig into the details of this enough. I believe the ideal solution is to change the expected argument in `age_ceiling()` to be `Union[float, int]`.

Reply:
...and perhaps also have `age_ceiling()` return a type of `Union[float, int]`? Although not including this change doesn't seem to cause an error.

Reply:
Ah, sorry I should have mentioned this: I get an error when I _only_ change the type in `age_ceiling()` to `Union[float, int]`
```
error: Incompatible return value type (got "Union[int, float]", expected "Optional[int]")
```

Reply:
That's odd.. the mypy tests pass for me when I only change the param type in `age_ceiling()` to `Union[float, int]`.

Reply:
I know that @joverlee521 pipenv update 'd a few weeks ago to get the same mypy behavior that we're seeing Jenkins. Maybe we all need to update.

Reply:
I ran `pipenv sync` recently so I don't think that's the cause.

Reply:
@kairstenfay Did you run `pipenv sync --dev`? Default `pipenv sync` does not include development dependencies like mypy.

Reply:
Good point. yes I did.

Reply:
I created a test branch using my recommendation and the Travis CI build didn't fail.
[See it here](https://github.com/seattleflu/id3c-customizations/runs/1027654962). 

(Ignore the unused import ;))

Reply:
Huh, my errors were due to the return value of `age_ceiling` not having the right type, not the parameter. I'll just update both so it reflects the truth. 

It was initially just set to `float` because in [PEP 484 type hints](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower) says:
> Rather than requiring that users write import numbers and then use numbers.Float etc., this PEP proposes a straightforward shortcut that is almost as effective: when an argument is annotated as having type float, an argument of type int is acceptable; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable.

Reply:
Hmm, I wonder if this is another mypy bug.

Reply:
I don't think this line is necessary anymore given your second commit, btw.

Reply:
It is required because the variable `string` is expected to be a string and cannot be reassigned to a float/int.

Reply:
But do we ever actually expect to get a non-string input variable here? We cast it to `int` before passing it to `age_ceiling`. Perhaps I'm missing something.